developer-docs-site/docs/concepts/events.md

```diff
@@ -3,24 +3,119 @@ title: "Events"
 slug: "events"
 ---
 
-Events are emitted during the execution of a transaction. Each Move module can define its own events and choose when to emit the events upon execution of the module. For example, during a [coin transfer](../tutorials/first-transaction.md), both the sender and receiver's accounts will emit `SentEvent` and `ReceivedEvent`, respectively. This data is stored within the ledger and can be queried via the REST interface's [Get events by event handle](https://fullnode.devnet.aptoslabs.com/v1/spec#/operations/get_events_by_event_handle).
+Events are emitted during the execution of a transaction. Each Move module can define its own events and choose when to
+emit the events upon execution of the module. Aptos Move starts with events with handle on mainnet launch, then switch
+to module events since v1.7.
 
-Assuming that an account `0xc40f1c9b9fdc204cf77f68c9bb7029b0abbe8ad9e5561f7794964076a4fbdcfd` had sent coins to another account, the following query could be made to the REST interface: `https://fullnode.devnet.aptoslabs.com/v1/accounts/c40f1c9b9fdc204cf77f68c9bb7029b0abbe8ad9e5561f7794964076a4fbdcfd/events/0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>/withdraw_events`. The output would be all `WithdrawEvent`s stored on that account, it would look like 
+# Module Events
+
+Since v1.7, Aptos Move released a new event framework, module events. Every event stream is identified by an event
+struct type instead of a handle.
+To define an event struct, just add attribute `#[event]` to a normal Move struct that has `drop + copy` abilities. For
+example,
+
+```rust
+/// 0xcafe::my_module_name
+/// An example module event struct denotes a coin transfer.
+#[event]
+struct TransferEvent has copy, drop {
+sender: address,
+receiver: address,
+amount: u64
+}
+```
+
+for example,
+
+```rust
+// Define an event.
+let event = TransferEvent {
+sender: 0xcafe,
+receiver: 0xface,
+amount: 100
+};
+// Emit the event just defined.
+0x1::event::emit(event);
+```
+
+The module event has been emitted.
+
+Example module events are available [here](https://explorer.aptoslabs.com/txn/682252266/events?network=testnet). Indices
+0, 1, 2 are three module events of
+type `0x66c34778730acbb120cefa57a3d98fd21e0c8b3a51e9baee530088b2e444e94c::event::MyEvent`. For module events compatible
+with the old events, `Account Address`, `Creation Number` and `Sequence Number` are all set to dummy value 0.
```

Comment:
Can we link to some indexer doc here to talk about how to access events off chain (via the events table available in hashura) + writing custom event processor? cc @rtso

Reply:
We can add something like:

To access events off-chain, you can query the Labs-Hosted GraphQL API: https://aptos.dev/indexer/api/labs-hosted#hasura-explorer
For complex event processing, you can write a custom event processor following this tutorial: https://github.com/aptos-labs/aptos-indexer-processors/tree/main/python/processors/coin_flip

Reply:
do we already have that yet?